src/settings.cpp

```diff
@@ -120,6 +415,268 @@ void read_settings(pugi::xml_node* root)
     };
     external_sources.push_back(std::move(source));
   }
+
+  // Check if we want to write out source
+  if (check_for_node(root, "write_initial_source")) {
+    write_initial_source = get_node_value_bool(root, "write_initial_source");
+  }
+
+  // Survival biasing
+  if (check_for_node(root, "survival_biasing")) {
+    survival_biasing = get_node_value_bool(root, "survival_biasing");
+  }
+
+  // Probability tables
+  if (check_for_node(root, "ptables")) {
+    urr_ptables_on = get_node_value_bool(root, "ptables");
+  }
+
+  // Cutoffs
+  if (check_for_node(root, "cutoff")) {
+    xml_node node_cutoff = root.child("cutoff");
+    if (check_for_node(node_cutoff, "weight")) {
+      weight_cutoff = std::stod(get_node_value(node_cutoff, "weight"));
+    }
+    if (check_for_node(node_cutoff, "weight_avg")) {
+      weight_survive = std::stod(get_node_value(node_cutoff, "weight_avg"));
+    }
+    if (check_for_node(node_cutoff, "energy_neutron")) {
+      energy_cutoff[0] = std::stod(get_node_value(node_cutoff, "energy_neutron"));
+    } else if (check_for_node(node_cutoff, "energy")) {
+      warning("The use of an <energy> cutoff is deprecated and should "
+        "be replaced by <energy_neutron>.");
+      energy_cutoff[0] = std::stod(get_node_value(node_cutoff, "energy"));
+    }
+    if (check_for_node(node_cutoff, "energy_photon")) {
+      energy_cutoff[1] = std::stod(get_node_value(node_cutoff, "energy_photon"));
+    }
+    if (check_for_node(node_cutoff, "energy_electron")) {
+      energy_cutoff[2] = std::stof(get_node_value(node_cutoff, "energy_electron"));
+    }
+    if (check_for_node(node_cutoff, "energy_positron")) {
+      energy_cutoff[3] = std::stod(get_node_value(node_cutoff, "energy_positron"));
+    }
+  }
+
+  // Particle trace
+  if (check_for_node(root, "trace")) {
+    auto temp = get_node_array<int64_t>(root, "trace");
+    trace_batch    = temp.at(0);
+    trace_gen      = temp.at(1);
+    trace_particle = temp.at(2);
```

Comment:
The error message for not specifying 3 numbers is very unhelpful.  Add something that explicitly checks the number of values and prints a helpful error message if it's wrong